mod-source-record-manager-server/src/main/java/org/folio/services/AbstractChunkProcessingService.java

```diff
@@ -56,6 +74,71 @@ public Future<Boolean> processChunk(RawRecordsDto incomingChunk, String jobExecu
         }).orElse(Future.failedFuture(new NotFoundException(String.format("Couldn't find JobExecution with id %s", jobExecutionId)))));
   }
 
+  private boolean isNotSupportedJobProfileExists(JobExecution jobExecution) {
+    boolean isExists = false;
+    List<ProfileSnapshotWrapper> childProfiles = jobExecution.getJobProfileSnapshotWrapper().getChildSnapshotWrappers();
+    if (!CollectionUtils.isEmpty(childProfiles)) {
+      isExists = isExistsMatchProfileToInstanceWithActionUpdateMarcBib(childProfiles);
+    }
+    return isExists;
+  }
+
+  private boolean isExistsMatchProfileToInstanceWithActionUpdateMarcBib(Collection<ProfileSnapshotWrapper> profileSnapshotWrapperList) {
+    for (ProfileSnapshotWrapper profileSnapshotWrapper : profileSnapshotWrapperList) {
+      if ((profileSnapshotWrapper.getContentType() == MATCH_PROFILE) && (isMatchingMarcBibToInstance(profileSnapshotWrapper))) {
+        ProfileSnapshotWrapper actionProfileSnapshotWrapper = getChildSnapshotWrapperByType(profileSnapshotWrapper, ACTION_PROFILE);
+        if (actionProfileSnapshotWrapper != null) {
+          ProfileSnapshotWrapper mappingProfileSnapshotWrapper = getChildSnapshotWrapperByType(actionProfileSnapshotWrapper, MAPPING_PROFILE);
+          if ((mappingProfileSnapshotWrapper != null) && (isMappingMarcBibToMarcBib(mappingProfileSnapshotWrapper))) {
+            return true;
+          }
+        }
+        isExistsMatchProfileToInstanceWithActionUpdateMarcBib(profileSnapshotWrapper.getChildSnapshotWrappers());
```

Comment:
I am fuzzy on the business rules but thinking this line should have a `return` here; some result passed back to the base case of your recursive approach. As is, if the line returns true or false, it does not seem to have any impact. Please correct me if I am wrong.

Reply:
Yes. You are absolutely right. Thank you. I lost return during a lot of code reviews. 
Fixed.